src/Element/TagElementListener.php

```
@@ -0,0 +1,30 @@
+final class TagElementListener
+{
+    public function __construct(
+        private readonly CacheTagCollector $tagCollector,
+        private readonly EventDispatcherInterface $dispatcher,
+    ) {
+    }
+
+    public function __invoke(ElementEventInterface $event): void
+    {
+        $taggingEvent = $this->dispatcher->dispatch(ElementTaggingEvent::fromElement($event->getElement()));
+        \assert($taggingEvent instanceof ElementTaggingEvent);
```

PHPStan (2.0.3)

Call to function assert() with true will always evaluate to true.
Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your %configurationFile%.

(function.alreadyNarrowedType)